addons/l10n_sa_edi/tests/common.py

```diff
@@ -213,7 +215,11 @@ def _create_invoice(self, **kwargs):
             }),
             ],
         }
-        move = self.env['account.move'].create(vals)
+        saudi_user = kwargs.get('saudi_user', False)
+        if saudi_user:
+            move = self.env['account.move'].with_user(self.user_saudi.id).create(vals)
+        else:
+            move = self.env['account.move'].create(vals)
```

Comment:
Instead of passing a flag, please pass the user.
```suggestion
        user = kwargs.get('user') or self.env.user
        move = self.env['account.move'].with_user(user.id).create(vals)
```
So if we need to add a new test with another user, we don't have to change the code :)
Btw, as default value I selected the `env.user` but it could be an empty recordset, `with_user` handles it well.